### PR TITLE
Adding BFSI Consent Management DAO Module Implementation

### PR DESCRIPTION
**Pull request by Ashi1993**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/org.wso2.bfsi.consent.management.dao/src/main/java/org/wso2/bfsi/consent/management/dao/queries/ConsentMgtCommonDBQueries.java [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+    public String getSearchExpiringConsentPreparedStatement(String statusesEligibleForExpirationCondition) {
+
+        return "SELECT OBC.CONSENT_ID " +
+                " FROM   OB_CONSENT_ATTRIBUTE CA " +
+                " JOIN   OB_CONSENT OBC " +
+                " ON     CA.CONSENT_ID = OBC.CONSENT_ID " +
+                " WHERE  CA.ATT_KEY = ? AND OBC.CURRENT_STATUS IN " + statusesEligibleForExpirationCondition;
+    }
+
+//    public String getGetConsentMappingResourcesForStatusPreparedStatement() {
```
Comment (hasithakn) [inline components/org.wso2.bfsi.consent.management.dao/src/main/java/org/wso2/bfsi/consent/management/dao/queries/ConsentMgtCommonDBQueries.java]: shall we remove these commented statements if these are not in use.

Comment (Ashi1993) [reply components/org.wso2.bfsi.consent.management.dao/src/main/java/org/wso2/bfsi/consent/management/dao/queries/ConsentMgtCommonDBQueries.java]: Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/2/commits/544143fd1d3ba2b3d581a41775d7e7bda95918ab

File: components/org.wso2.bfsi.consent.management.dao/pom.xml [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-collections4</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.bfsi.consent.management</groupId>
+            <artifactId>org.wso2.bfsi.consent.management.common</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <!-- Unit Test Dependencies -->
```
Comment (aka4rKO) [inline components/org.wso2.bfsi.consent.management.dao/pom.xml:65]: there are some non test dependencies under this comment

Comment (Ashi1993) [reply components/org.wso2.bfsi.consent.management.dao/pom.xml:65]: Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/2/commits/8251d91f68c97ed292b84e526bdca1e3358ef563

File: components/org.wso2.bfsi.consent.management.dao/src/main/java/org/wso2/bfsi/consent/management/dao/exceptions/ConsentDataUpdationException.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.bfsi.consent.management.dao.exceptions;
+
+/**
+ * OBConsentDataUpdationException.
```
Comment (hasithakn) [inline components/org.wso2.bfsi.consent.management.dao/src/main/java/org/wso2/bfsi/consent/management/dao/exceptions/ConsentDataUpdationException.java]:
```suggestion
 * ConsentDataUpdationException class
```

Comment (hasithakn) [reply components/org.wso2.bfsi.consent.management.dao/src/main/java/org/wso2/bfsi/consent/management/dao/exceptions/ConsentDataUpdationException.java]: Check other places as well

Comment (Ashi1993) [reply components/org.wso2.bfsi.consent.management.dao/src/main/java/org/wso2/bfsi/consent/management/dao/exceptions/ConsentDataUpdationException.java]: FIxed in https://github.com/wso2-extensions/bfsi-consent-management/pull/2/commits/9ef915919e894072ba2c880008eb847bc5db2fa6

**Review by hasithakn — APPROVED** [review]

**Review by anjuchamantha — DISMISSED** [review]

File: pom.xml [hunk 4]
```diff
@@ -218,12 +268,23 @@
...
+        <commons-lang.version>3.4</commons-lang.version>
+        <json.wso2.version>3.0.0.wso2v4</json.wso2.version>
+        <com.googlecode.json.version>1.1.1</com.googlecode.json.version>
+        <commons-collections.version>4.4</commons-collections.version>
+        <org.testng.version>7.7.0</org.testng.version>
+        <jacoco.version>0.8.6</jacoco.version>
+        <commons-dbcp.version>1.4</commons-dbcp.version>
+        <orbit.version.h2.engine>1.2.140.wso2v3</orbit.version.h2.engine>
+        <mockito.version>1.10.19</mockito.version>
+        <powermock.version>1.7.1</powermock.version>
```
Comment (aka4rKO) [inline pom.xml:287]: cant we use only mockito like we discussed offline?

Comment (Ashi1993) [reply pom.xml:287]: Powermock allows to mock static methods once per class. But in mockito we need to mock in each test. Hence used powermock to mock static methods.

**Review by hasithakn — APPROVED** [review]

**Review by Akila94 — APPROVED** [review]

**Review by aka4rKO — APPROVED** [review]

**Pull request merged**